celery_unique/core.py

```diff
@@ -0,0 +1,199 @@
+            # The new record will be given a TTL that allows it to expire
+            # (approximately) at the same time that the task is executed.
+            ttl = self._make_ttl_for_unique_task_record(options)
+            self.unique_backend.create_task_record(key, rv.task_id, ttl)
+
+        return rv
+
+    def _handle_as_unique(self, options):
+        """
+        Determines if a task should be handles as unique.
+
+        :param dict options: The options dict passed to `apply_async`.
+        """
+        return (
+            callable(self.unique_key)
+            and ('eta' in options or 'countdown' in options)
+            and self.unique_backend is not None
+        )
+
+    def _revoke_extant_task(self, key):
+        """
+        Removes an extant task
+
+        Not that this removed the key from the store regardless of whether the
+        task is still queued or not.
+        """
+        task_id = self.unique_backend.get_task_id(key)
+        if task_id is not None:
+            self.app.AsyncResult(task_id).revoke()
+            self.unique_backend.revoke_extant_task(key)
+
+    def make_key(self, callback_args, callback_kwargs):
+        """
+        Creates a key used to identify the task's unique configuration in Redis.
+
+        Note: All positional arguments and/or keyword arguments sent to the
+        task are applied identically to the task's bound `unique_key` callable.
+
+        :param callback_args: The positional arguments which will be passed to
+            the task when it executes @type callback_args: list | tuple
+        :param callback_kwargs: The keyword arguments which will be passed to
+            the task when it executes @type callback_kwargs: dict
+
+        :return: The key which will be used to find any extant version of this
+        task which, if found, will by revoked.  Keys are built by using three
+        colon-delimited components::
+
+            1. A global prefix used to identify that the key/value pair in
+                the backend was created to track
+                a unique Celery task (by default, this is "celery_unique")
+            2. The name of the task (usually the Python dot-notation path to
+                the function)
+            3. The value produced by the `key_generator` callable when supplied
+                with the task's callback arguments.
+
+        :rtype: unicode
+        """
+
+        # Get the unbound lambda used to create `self.unique_key` if the inner
+        # function exists
+        if hasattr(self.unique_key, '__func__'):
+            key_generator = self.unique_key.__func__
+        else:
+            key_generator = self.unique_key
+
+        # Create and return the redis key with the generated unique key suffix
+        return '{prefix}:{task_name}:{unique_key}'.format(
+            prefix=UNIQUE_KEY_PREFIX,
+            task_name=self.name,
+            unique_key=key_generator(*callback_args, **callback_kwargs),
```

Comment:
Following my other comment about the argument defaults for `UniqueTaskMixin.apply_async`, I think this will need to be changed to:
```python
unique_key=key_generator(
    *(callback_args or ()),
    **(callback_kwargs or {}),
)
```